sockeye/utils.py

```diff
@@ -142,6 +142,22 @@ def compute_lengths(sequence_data: mx.sym.Symbol) -> mx.sym.Symbol:
     return mx.sym.sum(sequence_data != C.PAD_ID, axis=1)
 
 
+def compute_length_ratio(input_seq: mx.nd.NDArray, label_seq: mx.nd.NDArray) -> mx.nd.NDArray:
```

Comment:
I like this idea! You don't even need an if-statement there. If you don't use the computed length_ratio later in the graph, it will be pruned from it anyway.

Reply:
Very good point! I didn't think of it. Trying out now.

Reply:
Implemented this, data_io is not modified now. Had to write a small override for ```update_dict``` in a custom metric to copy outputs to labels to make it work.